### PR TITLE
Adds Gather Subset Feature to na_ontap_gather_facts

### PR DESCRIPTION
**Pull request by carchi8py**
##### SUMMARY
Adds Gather Subset Feature to na_ontap_gather_facts
This was originally #47016

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
- na_ontap_gather_facts

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: test/units/modules/storage/netapp/test_na_ontap_gather_facts.py [hunk 1]
```diff
@@ -65,20 +66,49 @@ def invoke_successfully(self, xml, enable_tunneling):  # pylint: disable=unused-
...
+        elif self.type == 'zapi_error':
+            error = netapp_utils.zapi.NaApiError('test', 'error')
+            raise error
         self.xml_out = xml
-        print(xml.to_string())
         return xml
 
     @staticmethod
-    def build_vserver_info(vserver):
+    def build_vserver_info():
```
Comment (thedoubl3j) [inline test/units/modules/storage/netapp/test_na_ontap_gather_facts.py:79]: Is this code repeated below? specficially lines 89-96?

**Review by thedoubl3j — COMMENTED** [review]
additional test coverage and CI is looking good. only question is in my other comment.

File: test/units/modules/storage/netapp/test_na_ontap_gather_facts.py (hunk 1 above)
Comment (lonico) [reply test/units/modules/storage/netapp/test_na_ontap_gather_facts.py:79]: Yes, good catch.   We can delete one of the two functions, they are now identical.

Comment (thedoubl3j) [reply test/units/modules/storage/netapp/test_na_ontap_gather_facts.py:79]: 👍

**Review by lonico — APPROVED** [review]
shipit

**Comment by carchi8py** [comment]
@thedoubl3j We are not 100% sure why ansibot want us to rebase.

**Comment by thedoubl3j** [comment]
@carchi8py somewhere in your commit history there are commits such as commit  [4246983](https://github.com/ansible/ansible/commit/4246983e6b71d04a831acc5f5d41105557859957). Those commits failed CI so that I am assuming that is why ansibot wants you to rebase. ping @gundalow if you have any insights as to what else this could be. I am seeing no merge conflicts

**Comment by carchi8py** [comment]
This was rebased. @gundalow 
in branch devel i did a rebase
then i merged devel in to this branch.

Should i be rebaseing devel directly in to this branch instead?

**Comment by mattclay** [comment]
@carchi8py I pulled a copy of this PR locally to verify there is a merge conflict. The first conflict I encountered was in `lib/ansible/modules/storage/netapp/na_ontap_gather_facts.py`. That is why both GitHub and the bot are saying there are merge conflicts.

You'll need to rebase the commits from this branch on top of devel, rather than merging devel back into your branch. The [rebasing instructions](https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html) should be helpful in walking you through the process.

A rebase will result in replaying each commit from your branch on top of the latest version of the devel branch. If instead you see the last commit as `Merge branch 'devel' into 1296` then you've instead merged the devel branch into yours, rather than rebasing yours on top of the devel branch.

If you need further assistance with rebasing you can ask here, or for a quicker response, try asking in the #ansible-devel channel in IRC on Freenode.

**Comment by thedoubl3j** [comment]
thanks for the insight @mattclay added to our internal notes.

**Review by lonico — APPROVED** [review]
shipit

**Review by thedoubl3j — APPROVED** [review]
git conflict hell fixed. lgtm. shipit

**Pull request merged**